python/dgl/graph.py

```diff
@@ -1050,6 +1050,15 @@ def __init__(self,
         # set parent if the graph is a subgraph.
         self._parent = parent
 
+    def __setstate__(self, state):
+        # Compatibility with pickles from DGL 0.4.2-
+        if '_batch_num_nodes' not in state:
```

Comment:
My general comment is using `dgl.__version__` for if condition. Otherwise, LGTM.

Reply:
Decided not to do that since using `dgl.__version__` is not straightforward for nightly builds.